src/github.com/matrix-org/dendrite/clientapi/auth/storage/accounts_table.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2017 Vector Creations Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storage
+
+import (
+	"database/sql"
+	"fmt"
+	"time"
+
+	"github.com/matrix-org/dendrite/clientapi/auth/types"
+	"github.com/matrix-org/gomatrixserverlib"
+)
+
+const accountsSchema = `
+-- Stores data about accounts.
+CREATE TABLE IF NOT EXISTS accounts (
+    -- The Matrix user ID localpart for this account
+    localpart TEXT NOT NULL PRIMARY KEY,
+    -- When this account was first created, as a unix timestamp (ms resolution).
+    created_ts BIGINT NOT NULL,
+    -- The password hash for this account. Can be NULL if this is a passwordless account.
+    password_hash TEXT
+    -- TODO:
+    -- is_guest, is_admin, appservice_id, upgraded_ts, devices, any email reset stuff?
+);
+`
+
+const insertAccountSQL = "" +
+	"INSERT INTO accounts(localpart, created_ts, password_hash) VALUES ($1, $2, $3)"
+
+const selectAccountByPasswordHashSQL = "" +
+	"SELECT localpart WHERE localpart = $1 AND password_hash = $2"
```

Comment:
This shouldn't work since the password hash includes a random salt.

You'll need to read the password hash from the database and then call bcrypt.CompareHashAndPassword passing the hash and the password.